docs/by-example/sqlalchemy/inspection-reflection.rst

```diff
@@ -0,0 +1,122 @@
+=====================================================
+SQLAlchemy: Database schema inspection and reflection
```

Comment:
Is anything of this section CrateDB specific? Looks like it's mostly explaining SQLAlchemy, in which case we could refer to the upstream docs instead and test the functionality in python tests

Reply:
Hi. While the SQLAlchemy API methods per se are not specific to CrateDB, I found this section worth to keep, to display different kinds of inspection and reflection methods of SQLAlchemy together with their slightly CrateDB-specific outcomes, also because they are not mentioned elsewhere in our documentation (modulo `has_table`), and also because the corresponding spots are a bit obfuscated within the SQLAlchemy documentation.

In other words, while bringing those snippets together from different parts of our documentation, I have made an extra effort to discriminate between the different kinds of introspection/reflection features offered by SQLAlchemy:

- SQLAlchemy inspector
- Schema-supported reflection
- CrateDialect-specific reflection methods

The result is a reasonably concise overview about them, paired with CrateDB-specific responses.

```python
>>> inspector.get_schema_names()
['blob', 'doc', 'information_schema', 'pg_catalog', 'sys']

>>> inspector.default_schema_name
'doc'

>>> dialect.has_schema(connection, 'doc')
True
```